tests/kernel/timer/timer_api/src/main.c

```diff
@@ -61,9 +63,13 @@ static void duration_expire(struct k_timer *timer)
 
 	tdata.expire_cnt++;
 	if (tdata.expire_cnt == 1) {
-		TIMER_ASSERT(interval >= DURATION, timer);
+		TIMER_ASSERT((interval >= DURATION)
+			     || (INEXACT_MS_CONVERT
+				 && (interval == DURATION - 1)), timer);
```

Comment:
The inexact version is a tighter bound (looking for an exact interval value) than the old one (make sure it's higher than or equal to duration).  That seems wrong.  It's OK to add slop where needed, but we should be testing the same behavior on all platforms.

Reply:
I don't understand this statement.  If the original test passes nothing changes.  If it fails, then if we know we might have an error in converting ms to ticks, and in that case we relax the lower bound to account for a tick that wasn't added because accumulated error exceeded a tick.